lib/engine/game/base.rb

```diff
@@ -1375,16 +1376,17 @@ def ipo_reserved_name(_entity = nil)
         'IPO Reserved'
       end
 
-      def abilities(entity, type = nil, time: nil, owner_type: nil, strict_time: false)
+      def abilities(entity, type = nil, time: nil, on_phase: nil, passive_ok: nil)
```

Comment:
we don't need to make the change now, but i wanted to make note that the call to all_abilities is very expensive and is called quite often for garbage collection since it makes a ton of of new arrays

Reply:
Could we make a cache that's cleared after every action is processed? Any other ideas for ways to improve that?

Reply:
yea but it's difficult because abilities can change when companies are closed etc